gprofiler/profilers/factory.py

```diff
@@ -0,0 +1,38 @@
+from typing import TYPE_CHECKING, Any, List, Tuple, Union
+
+from gprofiler.log import get_logger_adapter
+from gprofiler.profilers.profiler_base import NoopProfiler
+
+if TYPE_CHECKING:
+    from gprofiler.main import UserArgs
+    from gprofiler.profilers.profiler_base import ProcessProfilerBase
+
+from gprofiler.profilers.perf import SystemProfiler
+from gprofiler.profilers.registry import get_profilers_registry
+
+logger = get_logger_adapter(__name__)
+
+
+def get_profilers(
+    user_args: 'UserArgs', **profiler_init_kwargs: Any
+) -> Tuple[Union['SystemProfiler', 'NoopProfiler'], List['ProcessProfilerBase']]:
+    profilers_registry = get_profilers_registry()
+    process_profilers_instances: List['ProcessProfilerBase'] = []
+    system_profiler: Union['SystemProfiler', 'NoopProfiler'] = NoopProfiler()
+    for profiler_name, profiler_config in profilers_registry.items():
+        profiler_mode = user_args.get(f"{profiler_name.lower()}_mode")
+        if profiler_mode in ("none", "disabled") or not user_args.get(profiler_name.lower(), True):
```

Comment:
Use your same helper method as used in `main.py` from that comment I gave earlier

Reply:
The one that check if a profiler is Noop?

Reply:
Seems irrelevant now since I can't find anything matching anymore. Perhaps I meant the Noop one